elixir/apps/api/test/api/controllers/identity_controller_test.exs

```diff
@@ -210,10 +299,91 @@ defmodule API.IdentityControllerTest do
         )
 
       assert resp = json_response(conn, 201)
+      assert resp["data"]["provider_identifier"] == attrs["provider_identifier"]
+      assert resp["data"]["email"] == attrs["provider_identifier"]
+    end
+
+    test "creates a resource with email attr only", %{
+      conn: conn,
+      account: account,
+      actor: api_actor
+    } do
+      {oidc_provider, _bypass} =
+        Fixtures.Auth.start_and_create_openid_connect_provider(account: account)
+
+      actor = Fixtures.Actors.create_actor(account: account)
+
+      attrs = %{"email" => "foo@localhost.local"}
+
+      conn =
+        conn
+        |> authorize_conn(api_actor)
+        |> put_req_header("content-type", "application/json")
+        |> post("/actors/#{actor.id}/providers/#{oidc_provider.id}/identities",
+          identity: attrs
+        )
+
+      assert resp = json_response(conn, 201)
+      assert resp["data"]["provider_identifier"] == attrs["email"]
+      assert resp["data"]["email"] == attrs["email"]
+    end
+
+    test "creates a resource with provider_identifier attr and email attr being the same value",
+         %{
+           conn: conn,
+           account: account,
+           actor: api_actor
+         } do
+      {oidc_provider, _bypass} =
+        Fixtures.Auth.start_and_create_openid_connect_provider(account: account)
 
+      actor = Fixtures.Actors.create_actor(account: account)
+
+      attrs = %{
+        "provider_identifier" => "foo@localhost.local",
+        "email" => "foo@localhost.local"
+      }
+
+      conn =
+        conn
+        |> authorize_conn(api_actor)
+        |> put_req_header("content-type", "application/json")
+        |> post("/actors/#{actor.id}/providers/#{oidc_provider.id}/identities",
+          identity: attrs
+        )
+
+      assert resp = json_response(conn, 201)
+      assert resp["data"]["provider_identifier"] == attrs["provider_identifier"]
+      assert resp["data"]["email"] == attrs["email"]
+    end
+
+    test "creates a resource with provider_identifier attr and email attr being different values",
```

Comment:
Might be worth having one more here that tests what happens when user provides input:

```
email: not a valid email
provider_identifier: nil
```

Reply:
Added a more explicit error test as recommended.